src/setfit/modeling.py

```diff
@@ -305,11 +313,11 @@ def _freeze_or_not(self, model: torch.nn.Module, to_freeze: bool) -> None:
         for param in model.parameters():
             param.requires_grad = not to_freeze
 
-    def predict(self, x_test: torch.Tensor) -> torch.Tensor:
+    def predict(self, x_test: Union[str, List[str]]) -> Union[torch.Tensor, "ndarray"]:
         embeddings = self.model_body.encode(x_test, normalize_embeddings=self.normalize_embeddings)
         return self.model_head.predict(embeddings)
 
-    def predict_proba(self, x_test: torch.Tensor) -> torch.Tensor:
+    def predict_proba(self, x_test: Union[str, List[str]]) -> Union[torch.Tensor, "ndarray"]:
```

Comment:
These two function signatures were misleading, since `SentenceTransformer.encode` takes a `Union[str, List[str]]`, so I updated it to be the same here.

Reply:
👍 this change was discussed in https://github.com/huggingface/setfit/pull/207#issuecomment-1330344680 as well.